### PR TITLE
amp-consent fullscreen interaction e2e

### PR DESCRIPTION
**Pull request by micajuine-ho**
Last task of #26432

Created an e2e test to test the fullscreen signal and interaction.

**Changes:**
- Created new `_test_` CMP (only available in test/dev mode) that points at the new `.html` file for the `promptUiSrc`
   - Access `_test_` CMP in the e2e test by declaring `__AMP_TEST = true` in `fullscreen-cmp.html`
- Minor change to `diy-consent.html` for clarity

**Notes:**
- Created new test fixtures (including the new _test_ CMP) instead of reusing example pages (and _ping_), so that we can have more control over what we test in the future
- In the e2e test, we are currently waiting 3s for the prompt to load, and 1 sec for the fullscreen signal and class to be applied. This seems to be working fine on Travis but might need to increase the sleep to avoid flakiness

### CODE REVIEW

File: examples/amp-consent/diy-consent.html [hunk 1]
```diff
@@ -199,10 +199,8 @@ <h1>Thank you!</h1>
     action: 'enter-fullscreen'
   };
 
-  console.log('diy-consent: Sending Fullscreen Message!', readyObject);
-
   parent.postMessage(readyObject, '*');
-}, 2500);
+}, 500);
```
Comment (zhouyx) [inline examples/amp-consent/diy-consent.html]: question: why attempting to set fullscreen every 500ms?

File: extensions/amp-consent/0.1/cmps.js [hunk 2]
```diff
@@ -28,12 +28,12 @@ import {getMode} from '../../../src/mode';
 export const CMP_CONFIG = {};
 
 if (getMode().test || getMode().localDev) {
-  CMP_CONFIG['_ping_'] = {
-    'consentInstanceId': '_ping_',
-    'checkConsentHref': '/get-consent-v1',
-    'promptUISrc': '/examples/amp-consent/diy-consent.html',
-  };
 }
+CMP_CONFIG['_ping_'] = {
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/cmps.js]: can we revert the change here? we don't want to ship the fake cmp

**Comment by mrjoro** [comment]
Is this PR still active?

File: extensions/amp-consent/0.1/cmps.js (hunk 2 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/cmps.js:32]: Need to find a way to work around this, as the e2e framework doesn't pick up the `ping` CMP b/c it's not in dev mode. Any ideas?

File: extensions/amp-consent/0.1/cmps.js [hunk 3]
```diff
@@ -34,6 +34,12 @@ if (getMode().test || getMode().localDev) {
     'promptUISrc':
       '/examples/amp-consent/diy-consent.html?cid=CLIENT_ID&pid=PAGE_VIEW_ID',
   };
+
+  CMP_CONFIG['_test_'] = {
+    'consentInstanceId': '_test_',
+    'checkConsentHref': '/get-consent-v1?cid=CLIENT_ID&pid=PAGE_VIEW_ID',
+    'promptUISrc': '/test/fixtures/served/cmp-iframe.html',
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/cmps.js:41]: could we override the `promptUISrc` with inline config, and get rid of either `_ping_` or `_test_`

File: extensions/amp-consent/0.1/test-e2e/test-cmp-fullscreen.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/**
```
Comment (zhouyx) [inline extensions/amp-consent/0.1/test-e2e/test-cmp-fullscreen.js:1]: I wonder if you want to add additional UI test here. For example the lightbox mode. If so I might rename the file. 
But that can also be a visual diff test. So your call

File: extensions/amp-consent/0.1/cmps.js (hunk 3 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/cmps.js:41]: Ahh yes, good point.

File: extensions/amp-consent/0.1/test-e2e/test-cmp-fullscreen.js (hunk 4 above)
Comment (micajuine-ho) [reply extensions/amp-consent/0.1/test-e2e/test-cmp-fullscreen.js:1]: I like the idea of doing more e2e tests, so I'll change the file name.
